samples/04-file-transfer/README.md

```diff
@@ -130,35 +180,99 @@ java -Dedc.fs.config=samples/04-file-transfer/consumer/config.properties -jar sa
-- the first query parameter (`connectorAddress`) is the address of the provider connector
-- the last query parameter (`destination`) indicates the desired _destination_ directory on you local machine.
-- `curl` will return the ID of the transfer process on the consumer connector.
+### 4. Request the file
 
-The consumer should spew out logs similar to:
+Now that we have a contract agreement, we can finally request the file. For this we use the new
+endpoint provided in the `api` extension of this sample. We specify the name of the file we want
+transferred as a path variable and provide the address of the provider connector, the path where
+we want the file copied and the contract agreement ID as query parameters:
 
 ```bash
-INFO 2021-09-07T17:24:42.128363 Received request for file test-document against provider http://localhost:8181/
-DEBUG 2021-09-07T17:24:42.592422 Request approved and acknowledged for process: 2b0a9ab8-78db-4753-8b95-77678cdd9fc8
-DEBUG 2021-09-07T17:24:47.425729 Process 2b0a9ab8-78db-4753-8b95-77678cdd9fc8 is now IN_PROGRESS
-DEBUG 2021-09-07T17:24:47.426115 Process 2b0a9ab8-78db-4753-8b95-77678cdd9fc8 is now COMPLETED
+curl -X POST "http://localhost:9191/api/file/test-document?connectorAddress=http://localhost:8181/api/ids/multipart&destination=/path/on/yourmachine&contractId={agreement ID}"
 ```
 
-and the provider should log something like:
+Again, we will get a UUID in the response. This time, this is the ID of the `TransferProcess`
+created on the consumer side. Because like the contract negotiation, the data transfer
+is handled in a state machine and performed asynchronously.
+
+Since transferring a file does not require any resource provisioning on either side, the transfer
+will be very quick and most likely already done by the time you read the UUID. 
+
+--- 
 
+You can also check the logs of the connectors to see that the transfer has been completed:
+
+Consumer side:
 ```bash
-DEBUG 2021-09-07T17:24:42.562909 Received artifact request for: test-document
-INFO 2021-09-07T17:24:42.568555 Data transfer request initiated
+INFO 2021-12-08T10:54:46.55678709 Received request for file test-document against provider http://localhost:8181/
+DEBUG 2021-12-08T10:54:47.454351767 Request approved and acknowledged for process: 98512dc2-3985-4696-937e-2c12c5ef77e3
+DEBUG 2021-12-08T10:54:52.123863179 Process 98512dc2-3985-4696-937e-2c12c5ef77e3 is now IN_PROGRESS
+DEBUG 2021-12-08T10:54:52.124975956 Process 98512dc2-3985-4696-937e-2c12c5ef77e3 is now COMPLETED
 ```
 
-then check `/path/on/yourmachine`, which should now contain a file named `test-document.txt`.
\ No newline at end of file
+Provider side:
+```bash
+DEBUG 2021-12-08T10:54:47.383011288 Received artifact request for: test-document
+INFO 2021-12-08T10:54:47.399136285 Data transfer request initiated
+INFO 2021-12-08T10:54:51.147997198 Destination path /path/on/yourmachine does not exist, will attempt to create
```

Comment:
This is the error message when you set a wrong path, but that should not be appear during the happy path

Reply:
As long as the destination file does not exist yet, I also got these logs when the transfer is successful.

Reply:
I saw this error message when I forgot to change the destination path, but when I set the right (and existing) path, that log didn't appear.

Reply:
If you set a non-existing path that can be created by the connector (I tried with a path on the Desktop) it appears, too. I just tested again without changing the /path/to/file and I get the following log:
```
Fatal error processing transfer request: 33ca945f-cbc1-4554-a3a7-cdfc3b72aa20. Error details: source file /path/to/file does not exist!
```